mlrun/track/base_tracker.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+from abc import abstractmethod
+
+import mlrun
+
+from ..utils import logger
+from .tracker import Tracker
+
+
+class BaseTracker(Tracker):
+    TRACKED_MODULE_NAME = ...
+
+    def __init__(self):
+        super().__init__()
+        self._tracked_platform = importlib.import_module(self.TRACKED_MODULE_NAME)
+        self._run_track_kwargs = {}
+        self._artifacts = {}
+
+    @classmethod
+    def is_enabled(cls) -> bool:
+        is_module_found = False
+        # Check for user configuration - defaulted to False if not configured:
+        if (
+            mlrun.mlconf.tracking.enabled
+            and getattr(mlrun.mlconf.tracking, cls.TRACKED_MODULE_NAME).mode
+            == "enabled"
+        ):
+            try:
+                # Check if the module is available - can be imported:
+                importlib.import_module(cls.TRACKED_MODULE_NAME)
+
+                is_module_found = True
+            except (ModuleNotFoundError, ImportError) as e:
+                logger.warning(
+                    f"Tracking enabled for {cls.TRACKED_MODULE_NAME}, but unable to import module "
+                    f"due to the following exception: {str(e)}"
+                )
```

Comment:
1. The general tracking enablement should be verified before even adding the trackers, there is no need to check if for every tracker.
2. I think this would be a better implementation with better explanation and error handling:
```suggestion
        # Check the tracker's configuration:
        if getattr(mlrun.mlconf.tracking, cls.TRACKED_MODULE_NAME).mode != "enabled":
            return False
        
        # Check if the module to track is available in the interpreter:
        try:
            importlib.import_module(cls.TRACKED_MODULE_NAME)
        except ModuleNotFoundError as module_not_found_error:
            # The module is missing in the interpreter:
            logger.warning(
                f"Tracking enabled for {cls.TRACKED_MODULE_NAME}, but the module was not found  "
                f"in the interpreter: {str(module_not_found_error)}"
            )
            return False
        except ImportError as import_error:
            # There was an error during the import of the module:
            logger.warning(
                    f"Tracking enabled for {cls.TRACKED_MODULE_NAME}, but MLRun was unable to import the "
                    f"module due to the following error: {str(import_error)}"
            )
        return True
```